### PR TITLE
[Mailer] Use `addPart` instead of `embed*` or `attach*`.

### LINKED ISSUE #17303: [Mime] Simplify adding Parts to an Email
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#47462
| PR author(s) | @fabpot
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by hiddewie**
Fixes https://github.com/symfony/symfony-docs/issues/17303
Ref https://github.com/symfony/symfony-docs/issues/17353
Ref https://github.com/symfony/symfony/pull/47711
Ref https://github.com/symfony/symfony/pull/47462

This PR contains the documentation for simplifications for adding parts to `Email`s.

### CODE REVIEW

File: mailer.rst [hunk 1]
```diff
@@ -446,22 +446,22 @@ result of rendering some template) or PHP resources::
...
     ;
 
-Alternatively you can use the ``attach()`` method to attach contents from a stream::
+Alternatively you can attach contents from a stream by passing it directly to the ``DataPart`` ::
 
     $email = (new Email())
         // ...
-        ->attach(fopen('/path/to/documents/contract.doc', 'r'))
+        ->addPart(new DataPart(fopen('/path/to/documents/contract.doc', 'r')))
     ;
```
Comment (mdoutreluingne) [inline mailer.rst:466]: End with a deprecated directive for version 6.2

File: mailer.rst [hunk 2]
```diff
@@ -1489,7 +1491,7 @@ FailedMessageEvent
     {
         // e.g you can get more information on this error when sending an email
         $event->getError();
-        
+
         // do something with the message
     }
```
Comment (mdoutreluingne) [inline mailer.rst:1510]: Same here for the Embedding Images section, ended with a deprecated directive for version 6.2

Comment (hiddewie) [reply mailer.rst:1510]: Done in 7f00524ba6d1e6562199e710ff8b32d304f4bbce

File: mailer.rst (hunk 1 above)
Comment (hiddewie) [reply mailer.rst:466]: Done in 7f00524ba6d1e6562199e710ff8b32d304f4bbce

**Comment by javiereguiluz** [comment]
Hidde, thanks for taking the time to update the mailer docs to use the new methods!

**Pull request merged**